Only emit changed settings.

sainthkh opened the pull request:
Closes #12038

----

- [x] There's a clear use-case for this code change => It's not causing any problem. But it can later.
- [x] Commit message has a short title & references relevant issues
- [x] The build will pass (run `yarn test` and `yarn lint`)

More info can be found by clicking the "guidelines for contributing" link above.

sainthkh committed:
Only emit changed settings.

issue #12038

allouis commented on core/server/services/settings/index.js:
```diff
@@ -12,11 +12,18 @@ module.exports = {
-            model.emitChange(model.attributes.key + '.' + 'edited', {});
+            const key = model.attributes.key;
+
+            if (newSettings[key] !== oldSettings[key]) {
```
This comparison is comparing object literals which will never be equal to each other.

I think this should be comparing the `value` of the settings!

sainthkh replied:
You're right. I fixed it and wrote unit tests for it.

sainthkh committed:
Compare the value of the value property.

ref #12038.

The type of setting is object. That's why we need to compare the value property.
And I tested this behavior with unit tests.

allouis commented:
Thank you @sainthkh :bow: :sun_with_face:
